specs-go/config.go

```diff
@@ -0,0 +1,49 @@
+package specs
+
+// Spec is the base configuration for CDI
+type Spec struct {
+	Version          string   `json:"cdiVersion"`
+	Kind             string   `json:"kind"`
+	KindShort        []string `json:"kindShort,omitempty"`
+	ContainerRuntime []string `json:"containerRuntime,omitempty"`
+
+	Devices        []Devices      `json:"devices"`
+	ContainerEdits ContainerEdits `json:"containerEdits,omitempty"`
+}
+
+// Devices is a "Device" a container runtime can add to a container
+type Devices struct {
+	Name           string         `json:"name"`
+	NameShort      []string       `json:"nameShort"`
+	ContainerEdits ContainerEdits `json:"containerEdits"`
+}
+
+// ContainerEdits are edits a container runtime must make to the OCI spec to expose the device.
+type ContainerEdits struct {
+	DeviceNodes []DeviceNode `json:"deviceNodes,omitempty"`
+	Hooks       []Hook       `json:"hooks,omitempty"`
+	Mounts      []Mount      `json:"mounts,omitempty"`
```

Comment:
Do we need an additional field to enforce ordering or we believe append would work?

Reply:
For now we can probably stay on an append only structure and try to infer the ordering. If a field is required we can reconsider when the use case presents itself